instrumentation/src/main.c

```diff
@@ -0,0 +1,99 @@
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdio.h>
+
+#define MAX_LINE_LENGTH 1023
+#define MAX_LINES 50
+
+#define ALLOWED_ENV_VARS "OTEL_SERVICE_NAME", "OTEL_EXPORTER_OTLP_ENDPOINT", "OTEL_RESOURCE_ATTRIBUTES", "SPLUNK_PROFILER_ENABLED", "SPLUNK_PROFILER_MEMORY_ENABLED", "SPLUNK_METRICS_ENABLED", "JAVA_TOOL_OPTIONS", "NODE_OPTIONS"
```

Comment:
Are you intending to rely on the conf contents for `JAVA_TOOL_OPTIONS` and `NODE_OPTIONS`?

Reply:
The contents of these conf files are required for other auto-instrumentation implementation such as chart/operator implementations. I think these will be required, can supply more details if needed.

Reply:
these are definitely required or you cannot inject java or nodejs agents.